### ISSUE TITLE
Use TypeScript 1.8

### ISSUE THREAD

**Issue by pietro909** [TypeStrong/typedoc#210]
Hi there, I'm having this issue:

![typedoc issue](https://cloud.githubusercontent.com/assets/2094604/15182985/6439c518-178f-11e6-86a0-b309beb43515.jpg)

While using [string literal type](https://basarat.gitbooks.io/typescript/content/docs/types/stringLiteralType.html) which are supported from TypeScript 1.8

    export type MyType = "One" | "Two" | "Three" | "Four";

Is there any plan to upgrade to 1.8, or a simple way to tell the plugin to pick up a different version?

Thanks

**Comment by aciccarello** [comment]
See #204 ([comment](https://github.com/TypeStrong/typedoc/pull/204#issuecomment-217676411)). I think it just needs to get pushed to npm

**Comment by DaSchTour** [comment]
I guess that won't be enough if there is no version published here?

**Comment by pietro909** [comment]
Yes, thank you: the problem is that working in a production pipeline I need a release, can't just put it manually there.
Have you got any plan for the next one?

**Comment by tht13** [comment]
The repo was recently adopted from the original creator into a larger community project (TypeStrong), we are now waiting for npm publish permisions from the same user. Hopefully this won't take too long :smile:

**Comment by pietro909** [comment]
Thank you!

**Comment by jocull** [comment]
I have a pull request for this: #213

**Comment by felixfbecker** [comment]
Maybe it would be beneficial to setup [greenkeeper](https://greenkeeper.io/) for this project. It would do an automatic PR every time a new TypeScript version is released and that PR would automatically get built by Travis so you could see if it breaks anything. If not, you can quickly merge it, so we can all use the latest and greatest TypeScript.

**Comment by WasabiFan** [comment]
@felixfbecker I believe that TypeDoc uses a version of TypeScript that is compiled manually instead of downloaded from npm because it needs some custom modifications to make internal compiler stuff public. I believe that Greenkeeper only operates on npm dependencies, so I don't think this would work.

It might work if TypeDoc were modified to use ntypescript instead of a custom build, but I'm not familiar with the architecture myself so I don't know if that's viable.

**Comment by felixfbecker** [comment]
@WasabiFan But it lists `typescript` as dependency in package.json

**Comment by WasabiFan** [comment]
I think that's rightfully a dev dependency for compiling typedoc itself, based on some discussion from a few days ago.

**Comment by WasabiFan** [comment]
I'm saying that I believe that the npm module that is downloaded is only used by a developer who is compiling typedoc itself. Once typedoc has been compiled, it uses its own version of the compiler that is separate.

**Comment by DaSchTour** [comment]
That's not true and that's the point way everyone is waiting for a release with TypeScript 1.8
Typedoc uses the typescript compiler for generating the docs. And because of the old version of TypeScript some projects can't use typedoc because the project will not compile and no docs are created.

**Comment by WasabiFan** [comment]
> That's not true and that's the point way everyone is waiting for a release with TypeScript 1.8

I'm not really sure what you are saying. As I think you said, because TypeDoc doesn't currently use TypeScript 1.8, some projects that use newer language features can't be documented. But what are you saying isn't true? And who are you responding to?

**Comment by electricessence** [comment]
Wish this would work now. :(

**Comment by amejias101** [comment]
@electricessence you can clone and build the directory locally. Once you build it, you can `cd` into the directory and do a `npm install . -g` and your local copy will be accessible globally

**Comment by jocull** [comment]
To patch this in the meantime, I pushed up a repo with the compiled dependencies included. You can't add a reference to the official TypeDoc git repo because npm won't run the build step.

https://github.com/jocull/typedoc
https://github.com/jocull/grunt-typedoc

In package.json:

```
    "grunt-ts": "^5.4.0",
    "grunt-typedoc": "https://github.com/jocull/grunt-typedoc.git#master",
    "load-grunt-tasks": "^3.5.0",
```

Don't count on it being there for long, but it might get your builds running in the meantime (our doc generator was crashing and stopping us from moving forward)

**Comment by pietro909** [comment]
thanks @jocull 
though I eventually had to get rid of newest features and commit _the old way_ :(

**Comment by blakeembrey** [comment]
Released `0.4.0` to NPM.
